forcedotcom/packaging
# feat: make `SfProject` optional

This PR updates some packaging classes to make `SfProject` optional.

See: https://github.com/salesforcecli/plugin-packaging/pull/771

Most of these classes were taking an `SfProject` instance along with a possible package ID **or** alias (defined in sfdx-project.json). By making `SfProject` optional, consumers of the library can call it without having to create a dummy sfdx-project.

Note for review:
there are classes/methods that do need a project b/c they write stuff on it after a package operation, most of these aren't affected by this change (`PackageVersion.updateProjectWithPackageVersion` was updated to throw if there's no project).

QA:

Check the plugin-packaging PR linked above, with this branch linked (or the prerelease) you should be able to use the listed commands outside a project.

@W-16286213@
